src/Compilers/CSharp/Portable/Lowering/DiagnosticsPass_Warnings.cs

```diff
@@ -859,6 +859,39 @@ private bool CheckForAssignmentToSelf(BoundAssignmentOperator node)
             return false;
         }
 
+        private void CheckForDeconstructionAssignmentToSelf(BoundTupleLiteral leftTuple, BoundExpression right)
+        {
+            while (right.Kind == BoundKind.Conversion)
+            {
+                right = ((BoundConversion)right).Operand;
+            }
+
+            if (right.Kind != BoundKind.ConvertedTupleLiteral && right.Kind != BoundKind.TupleLiteral)
+            {
+                return;
+            }
+
+            var rightTuple = (BoundTupleExpression)right;
+            var leftArguments = leftTuple.Arguments;
+            int length = leftArguments.Length;
+            Debug.Assert(length == rightTuple.Arguments.Length);
+
+            for (int i = 0; i < length; i++)
+            {
+                var leftArgument = leftArguments[i];
+                var rightArgument = rightTuple.Arguments[i];
+
+                if (leftArgument.Kind == BoundKind.TupleLiteral)
```

Comment:
I don't think we need to.
The left-hand-side of deconstruction is purely composed of nested `BoundTupleLiteral`s, with no conversions. See `DeconstructionVariablesAsTuple`.